PSJira/Public/Add-JiraIssueWorklog.ps1

```diff
@@ -0,0 +1,143 @@
+        [String] $Comment,
+
+        # Issue that should be commented upon
+        [Parameter(Mandatory = $true,
+                   Position = 1,
+                   ValueFromPipeline = $true,
+                   ValueFromPipelineByPropertyName = $true)]
+        [Alias('Key')]
+        [Object] $Issue,
+
+        # Time spent to be logged, in seconds
+        [Parameter(Mandatory = $true,
+                   Position = 2,
+                   ValueFromPipeline = $true,
+                   ValueFromPipelineByPropertyName = $true)]
+        [Int] $TimeSpent,
+
+        # Date/time started to be logged
+        [Parameter(Mandatory = $true,
+                   Position = 3,
+                   ValueFromPipeline = $true,
+                   ValueFromPipelineByPropertyName = $true)]
+        [String] $DateStarted,
+
+        # Visibility of the comment - should it be publicly visible, viewable to only developers, or only administrators?
+        [ValidateSet('All Users','Developers','Administrators')]
+        [String] $VisibleRole = 'Developers',
+
+        # Credentials to use to connect to Jira. If not specified, this function will use
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential] $Credential
+    )
+
+    begin
+    {
+        Write-Debug "[Add-JiraIssueWorklog] Begin"
+        # We can't validate pipeline input here, since pipeline input doesn't exist in the Begin block.
+    }
+
+    process
+    {
+#        Write-Debug "[Add-JiraIssueWorklog] Checking Issue parameter"
+#        if ($Issue.PSObject.TypeNames[0] -eq 'PSJira.Issue')
+#        {
+#            Write-Debug "[Add-JiraIssueWorklog] Issue parameter is a PSJira.Issue object"
+#            $issueObj = $Issue
+#        } else {
+#            $issueKey = $Issue.ToString()
+#            Write-Debug "[Add-JiraIssueWorklog] Issue key is assumed to be [$issueKey] via ToString()"
+#            Write-Verbose "Searching for issue [$issueKey]"
+#            try
+#            {
+#                $issueObj = Get-JiraIssue -Key $issueKey -Credential $Credential
+#            } catch {
+#                $err = $_
+#                Write-Debug 'Encountered an error searching for Jira issue. An exception will be thrown.'
+#                throw $err
+#            }
+#        }
+#
+#        if (-not $issueObj)
+#        {
+#            Write-Debug "[Add-JiraIssueWorklog] No Jira issues were found for parameter [$Issue]. An exception will be thrown."
+#            throw "Unable to identify Jira issue [$Issue]. Does this issue exist?"
+#        }
+
+        Write-Debug "[Add-JiraIssueWorklog] Obtaining a reference to Jira issue [$Issue]"
+        $issueObj = Get-JiraIssue -InputObject $Issue -Credential $Credential
+
+        $url = "$($issueObj.RestURL)/worklog"
+
+        Write-Debug "[Add-JiraIssueWorklog] Creating request body from comment"
+        $props = @{
+            'comment' = $Comment;
+            'started' = $DateStarted;
```

Comment:
Would require change (naming the desired property) if the type is changed, as suggested above